bliss/encoder/convnet.py

```diff
@@ -66,7 +66,8 @@ def __init__(self, n_bands, ch_per_band, num_features, double_downsample=True):
             nn.Sequential(*[ConvBlock(64, 64, kernel_size=5, padding=2) for _ in range(4)]),
             ConvBlock(64, 128, stride=2),
             nn.Sequential(*[ConvBlock(128, 128) for _ in range(5)]),
-            ConvBlock(128, num_features, stride=(2 if double_downsample else 1)),  # 4
+            nn.Sequential(*[ConvBlock(128, 128, stride=2) for _ in range(num_downsample)]),
+            ConvBlock(128, num_features, stride=1),  # 4
```

Comment:
A potential issue here is that the downsampling doesn't happen until several layers into the network, which sometimes is what you want, but other times isn't. For high-resolution images, you likely want to downsample right away (and to set the kernel size to 5 rather than 3), to reduce memory usage.

Reply:
Pushed all the downsampling layers to the front and increased kernel size to 5. Please let me know if this works!

Reply:
Unfortunately not, because this change will break other case studies. Maybe two binary flags is the way to go: one for down sampling here and the other for down sampling right away

Reply:
Do we want to allow the possibility of downsampling at both places? or just one of the two depending on the flag?

Reply:
I'm not really sure, either way seems OK for now

Reply:
I added a `downsample_at_front` flag. Seems like the network would have one common backbone, and the flag either puts the downsampling layers at the front or the back, followed by upsampling back to `num_features`. Default value is `false` based on what works for other case studies I think. Added the flag to hydra config as well.

Reply:
Resolved for now, things to look into next:
- Upscale number of channels per downsampling layer
- Experiment with attention blocks
- Intersperse successive downsampling layers with C3 blocls